zachsinterview/src/main/java/com/imprev/zachsinterview/NumericalKeyValueFunImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.imprev.zachsinterview;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class NumericalKeyValueFunImpl implements NumericalKeyValueFun {
+    
+    private NumericalKeyValueStore keyValueStore;
+    
+    public NumericalKeyValueFunImpl(NumericalKeyValueStore keyValueStore) {
+        this.keyValueStore = keyValueStore;
+    }
+
+    @Override
+    public Integer sumByPrefix(String prefix) {
+        
+        int sum = 0;
+        try {
+            Iterator<String> iter = this.keyValueStore.getKeysByPrefix(prefix);
+            List<String> allKeys = new ArrayList<String>();
+            iter.forEachRemaining(allKeys::add);
+            
+            for (String key : allKeys) {
+                sum += this.keyValueStore.get(key);
+            }
+            
+        } catch (NumericalKeyValueException e) {
+            e.printStackTrace();
```

Comment:
We should include logging here, rather than just printing the stack trace. Additionally, we should throw an exception to indicate that there was a failure. currently we just fail silently. 

For example
catch (NumericalKeyValueException e) {
            logger.error(e.printStackTrace());
            throw new Exception("Exception with prefix " + prefix, e);
}